src/components/ruler/Numbers.tsx

```diff
@@ -1,39 +1,35 @@
-import { useEffect } from "react";
+import { useEffect, useState } from "react";
 import { useNumbersLineContext } from "../../context/numbersLineContext";
 import { LineRange } from "../../type/Line";
-
 interface IProps {
-  startIndex: number;
-  setStartIndex: (val: number) => void;
+  leftPosition: number;
 }
-
-const Numbers = ({ startIndex, setStartIndex }: IProps) => {
+const Numbers = ({ leftPosition }: IProps) => {
   const { kind } = useNumbersLineContext();
-  var endIndex = kind == LineRange.hundredCircular ? 101 : 21;
-
-  var labels = [];
-  if (kind == LineRange.hundredCircular)
-    labels = Array.from({ length: kind + 1 }, (_, index) => (index % 10 == 0 ? { value: index, isMainLine: true } : { value: index, isMainLine: false }));
-  else labels = Array.from({ length: kind }, (_, index) => ({ value: index, isMainLine: true }));
+  const [labels, setLabels] = useState<number[]>([]);
 
   useEffect(() => {
-    setStartIndex(0);
+    let array: number[] = [];
+    array = Array.from({ length: kind == LineRange.hundred ? kind + 1 : kind }, (_, index) => index);
+    setLabels(array);
   }, [kind]);
+
   return (
-    <>
-      {labels.slice(startIndex, startIndex + endIndex).map(({ value, isMainLine }) =>
-        isMainLine ? (
-          <div className="flex flex-col items-center">
+    <div
+      className="fixed left-0 right-0 flex justify-between border-t-4 border-gray-900 pt-0 mx-0 pl-8 pr-8"
+      style={kind == LineRange.hundred ? { width: window.innerWidth * 4.8, left: `${leftPosition}px` } : {}}
```

Comment:
calc 4.8 like : 100/rulerLength.hundred * window.innerWidth
and then put this var instead